manifest/yaml.go

```diff
@@ -271,41 +272,29 @@ func (pp *Ports) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*pp = make(Ports, len(v))
 
 	for i, s := range v {
-		parts := strings.Split(s, ":")
-		p := Port{}
-
-		switch len(parts) {
-		case 1:
-			n, err := strconv.Atoi(parts[0])
-
-			if err != nil {
-				return fmt.Errorf("error parsing port: %s", err)
-			}
-
-			p.Name = parts[0]
-			p.Container = n
-			p.Balancer = n
-			p.Public = false
-		case 2:
-			n, err := strconv.Atoi(parts[0])
-
-			if err != nil {
-				return fmt.Errorf("error parsing port: %s", err)
-			}
+		parts := portMappingRegex.FindStringSubmatch(s)
+		if len(parts) == 0 {
+			return fmt.Errorf("invalid portmapping %s", s)
+		}
 
-			p.Balancer = n
+		p := Port{}
+		p.Name = parts[1]
 
-			n, err = strconv.Atoi(parts[1])
+		protocol := strings.ToLower(parts[3])
+		if protocol != string(TCP) && protocol != string(UDP) {
+			protocol = string(TCP)
+		}
+		p.Protocol = Protocol(protocol)
 
-			if err != nil {
-				return fmt.Errorf("error parsing port: %s", err)
-			}
+		container, _ := strconv.Atoi(parts[2])
+		p.Container = container
+		p.Balancer = container
 
-			p.Name = parts[0]
-			p.Container = n
+		// Only TCP ports can be "public" (in the ELB sense) or have an ELB at all
+		if parts[1] != "" && p.Protocol == TCP {
```

Comment:
Sorry, just to clarify, you think it makes sense to go back and change it, or to leave it as is?